kube/pod.go

```diff
@@ -252,301 +196,205 @@ func getEnvVars(role *model.Role, defaults map[string]string, secrets SecretRefM
 			var ok bool
 			ok, stringifiedValue = config.Value(defaults)
 			if !ok {
+				// XXX When is this valid?  If never, why don't we throw an error?
```

Comment:
Yeah, but how can this happen. Doesn't every variable have at least the empty string as the default?  Can you give an example?

Reply:
I don't think so: if there's no `default:` at in the role manifest, it would be nil (rather than empty), right? Things like `DOMAIN`… Though I think in the original case it was something like having an unset environment variable vs. having one set to empty string, triggering different behaviour in the bash scripts later on.

Reply:
I thought `Default` would be a `string`, so couldn't be `nil`, but it turns out to be an `interface{}`.  I've updated the comment in https://github.com/SUSE/fissile/commit/345a015c2d75aec4a02d8409ad458793f308aa83